### PR TITLE
First try at composition in c++

### PR DESCRIPTION
**Pull request by JakobRoseke**
It does not work at the moment, it complains in 18 of Webserver.h that 'WebServer' does not name a type.

### CODE REVIEW

File: src/smartCar/Webserver.h [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+
+//INITIALIZE THE WEBSERVER
+void webserverInit() {}
+
+//CREATE AND UPDATE THE WEBSERVER
+void webserverCreation() {}
+
+void handle_OnConnect() {}
+
+void handle_NotFound() {}
```
Comment (platisd) [inline src/smartCar/Webserver.h:36]: Move these inside the class declaration and without the `{}` part.
E.g.: `void webserverCreation();`

**Comment by JakobRoseke** [comment]
> There are multiple issues with this, please open a pull request and invite me for a review there.

Sure, I am not entirely sure if I want to merge this code as the current code in the master branch works. Even if it is a little ugly it still works and should be fine for this project.

File: src/smartCar/Webserver.h [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
 
-extern WiFiClient client;
+String sendHTML() {}
+  
+};
```
Comment (platisd) [inline src/smartCar/Webserver.h:40]: Remove if you don't need

File: src/smartCar/Webserver.h [hunk 3]
```diff
@@ -1,9 +1,43 @@
...
+#include <ESPmDNS.h>
+
+class Webserver {
+
+  private:
+
+//VARIABLES
+const char* ssid;
+const char* password;
+WebServer server();
```
Comment (platisd) [inline src/smartCar/Webserver.h:18]: You probably should specify the argument. I'd suggest something like `WebServer server{80};`

File: src/smartCar/Webserver.h [hunk 4]
```diff
@@ -1,9 +1,43 @@
...
 #include <WiFi.h>
+#include <ESPmDNS.h>
+
+class Webserver {
+
+  private:
+
+//VARIABLES
+const char* ssid;
+const char* password;
```
Comment (platisd) [inline src/smartCar/Webserver.h:17]: Remove from header file and put in an anonymous namespace in the `.cpp` file.

```cpp
namespace {
const char* ssid = "bllasda";
const char* password = "blabla";
}
```

File: src/smartCar/Webserver.h [hunk 5]
```diff
@@ -1,9 +1,43 @@
...
 #define WEBSERVER_H
 
+
+#include <iostream>
+#include <HTTP_Method.h>
+#include <WebServer.h>
 #include <WiFi.h>
+#include <ESPmDNS.h>
+
+class Webserver {
```
Comment (platisd) [inline src/smartCar/Webserver.h:11]: Inside the class you should have only things that you want to publicly expose or things you want to privately share among your functions.

File: src/smartCar/smartCar.ino [hunk 6]
```diff
@@ -7,6 +7,8 @@
 //MANUAL (true) OR AUTOMATIC (false) MODE
 boolean nu = true;
 
+Webserver webserver;
+
```
Comment (platisd) [inline src/smartCar/smartCar.ino:11]: Have this here in one place, not both.

**Review by platisd — CHANGES_REQUESTED** [review]
I cannot comment on the `.cpp` file because it was not changed in this pull request. However, there are problems there as well.

Overall, you should look into [this](https://github.com/DIT112-V20/kotlin-app-arduino-sketch-ci/blob/master/arduino/SmartcarWebserverWithClass/SmartCarWebController.h) example on how the class should more or less look like.

Add the file again (e.g. a copy of it with a different name) if you want me to be able to see it and provide comments.

**Pull request closed without merging**